server/src/routers/admin/profile-years.ts

```diff
@@ -53,10 +69,26 @@ router.put('/:year', acl('profile-years:write'), async (req, res) => {
 });
 
 router.patch('/:year', acl('profile-years:write'), async (req, res) => {
-  await db<YearRecord>('app.years')
-    .where('profile_id', req.params.profile)
-    .andWhere('year', Number(req.params.year))
-    .update(req.body);
+  const updateData: Partial<YearRecord> = {};
```

Comment:
Dá se nějak jednoduše vyfiltrovat z `req.body` jen parametry které jsou přítomny v interface YearRecord @srdecny ? Tenhle můj pokus mi přijde dost neobratný

Reply:
mhm, kdyby se tam nemusely předělávat ty názvy proměnných z camelCase na snake_case, tak by to možná šlo elegantněji. takto mě napadá akorát něco jako tohle: https://gist.github.com/srdecny/3991cf1560dc7e1c9d83448afb8ce9a9, ale je to ošklivé jako hřích, takže mě tam ty ify ani moc nevadí. obecně je problém že v runtime se nedají moc elegnatně ověřovat typy (protože při transpilaci do js typy zmizí)

Reply:
Osobne mi cela ta konverze prijde jako zbytecnost. Sel bych podle databaze se vzorem "snake-case" a camel case z prace s daty bych vsude prevedl do snake case.